### PR TITLE
Add more unit tests on api/client/ps package

### PR DESCRIPTION
**Pull request by vdemeester**
Add some unit tests on `api/client/ps` 🐙. I've reorganized a little bit the code : moved `customFormat` to `formatter.go` as it seems the right place.

``` bash
# before
+ go test -test.timeout=60m github.com/docker/docker/api/client/ps
PASS
coverage: 59.1% of statements
# after
+ go test -test.timeout=60m github.com/docker/docker/api/client/ps
PASS
coverage: 100.0% of statements
```

There is a tests that does seems weird, I'll have to take a closer look.. 🐸 

Signed-off-by: Vincent Demeester vincent@sbr.pm

### CODE REVIEW

File: api/client/ps/formatter_test.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+			Context{
+				Format: "raw",
+				Output: out,
+			},
+			`container_id: containerID1
+image: ubuntu
+command: ""
+created_at: 1970-01-01 00:00:00 +0000 UTC
+status: 
+names: oobar_baz
```
Comment (vdemeester) [inline api/client/ps/formatter_test.go]: This doesn't look right at all.. I'm trying to see if it's my test(s) that are messing things up or the code.. :sweat:

File: api/client/ps/formatter_test.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+				Format: "raw",
+				Output: out,
+				Size:   true,
+			},
+			`container_id: containerID1
+image: ubuntu
+command: ""
+created_at: 1970-01-01 00:00:00 +0000 UTC
+status: 
+names: obar_baz
```
Comment (vdemeester) [inline api/client/ps/formatter_test.go]: This doesn't look right at all.. I'm trying to see if it's my test(s) that are messing things up or the code.. :sweat:

File: api/client/ps/formatter.go [hunk 3]
```diff
@@ -71,3 +76,65 @@ func tableFormat(ctx Context, containers []types.Container) {
 
 	customFormat(ctx, containers)
 }
+
+func customFormat(ctx Context, containers []types.Container) {
```
Comment (tiborvass) [inline api/client/ps/formatter.go]: @vdemeester did you just move this here or did you make some changes? (sorry lazy to check)

Comment (vdemeester) [inline api/client/ps/formatter.go]: > I've reorganized a little bit the code : moved customFormat to formatter.go as it seems the right place.

Just moved :wink:

File: api/client/ps/formatter_test.go [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+		context  Context
+		expected string
+	}{
+		// Errors
+		{
+			Context{
+				Format: "{{InvalidFunction}}",
+				Output: out,
+			},
+			"Template parsing error: template: :1: function \"InvalidFunction\" not defined\n",
```
Comment (tiborvass) [inline api/client/ps/formatter_test.go]: feel free to use backticks to avoid escaping `"`

Comment (vdemeester) [inline api/client/ps/formatter_test.go]: `true ` :sweat_smile:

**Comment by jfrazelle** [comment]
LGTM

**Comment by calavera** [comment]
LGTM

**Pull request merged**